### DISCUSSION TITLE
Transform for one-hot encoded Labels

### DISCUSSION THREAD

**Discussion by simon-forb** [pyg-team/pytorch_geometric#9468] [Ideas]
Hi,

Would something like this be of interest to PyG? I am using this for my own projects, but maybe it could be valuable to others.

```python
from torch_geometric.data import Data
from torch_geometric.transforms import BaseTransform
from torch.nn.functional import one_hot


class OneHotLabel(BaseTransform):
    def __init__(self, num_classes: int):
        self.num_classes = num_classes

    def forward(self, data: Data) -> Data:
        data.y = one_hot(data.y, self.num_classes).float()
        return data
```

**Comment by rusty1s** [comment]
What's the use-case for it? Since I don't see this is really related to GNNs, I don't think we have to necessarily integrate this into PyG.

**Reply by simon-forb** [reply]
This is used quite frequently in the graph mixup domain, e.g.:

Ma, Xinyu, et al. "Fused gromov-wasserstein graph mixup for graph-level classifications." Advances in Neural Information Processing Systems 36 (2024).

Han, Xiaotian, et al. "G-mixup: Graph data augmentation for graph classification." International Conference on Machine Learning. PMLR, 2022.

Zeng, Zhichen, et al. "Graph Mixup on Approximate Gromov–Wasserstein Geodesics." Forty-first International Conference on Machine Learning.

If you do not think this is valuable to PyG, I am totally fine with it. Just wanted to make an offer :)

**Reply by rusty1s** [reply]
Sorry for late reply. I think we can add it, but it is of lower-priority since we already provide `utils.one_hot`.
